spec/lib/withdraw_and_redirect_manual_spec.rb

```diff
@@ -33,30 +33,47 @@
     allow(publishing_adapter).to receive(:unpublish_and_redirect_manual_and_sections)
   end
 
-  it "withdraws the manual" do
+  it "withdraws the manual and unpublishes" do
     subject.execute
 
     reloaded_manual = Manual.find(manual.id, user)
     expect(reloaded_manual.withdrawn?).to eq(true)
-  end
-
-  it "calls the publishing adapter to unpublish the manual" do
-    subject.execute
     expect(publishing_adapter).to have_received(:unpublish_and_redirect_manual_and_sections)
-      .with(instance_of(Manual),
-            redirect:,
-            include_sections:,
-            discard_drafts:)
+                                    .with(instance_of(Manual),
+                                          redirect:,
+                                          include_sections:,
+                                          discard_drafts:)
   end
 
-  context "when there is no published manual" do
+  context "when the manual is in draft" do
     let(:state) { "draft" }
 
     it "raises an error" do
       expect { subject.execute }.to raise_error(WithdrawAndRedirectManual::ManualNotPublishedError)
     end
   end
 
+  context "when the manual is withdrawn" do
+    let(:state) { "published" }
+
+    before do
+      manual.withdraw
+      manual.save!(user)
+    end
+
+    it "withdraws and unpublishes the manual again" do
+      subject.execute
+
+      reloaded_manual = Manual.find(manual.id, user)
```

Comment:
The setup for this test feels a little bit awkward - could we create a separate context for each scenario (manual publisher/manual already withdrawn) rather than overriding the default context? It would make the tests easier to understand, I feel

Reply:
Kept the state let at the top as it feels somewhat "default" that it's set to published. The last test on dry-run also expects a default state. Removed the before so it reads better, and the let for our test as we were using the default anyway.